components/server/src/server.ts

```diff
@@ -172,7 +177,7 @@ export class Server<C extends GitpodClient, S extends GitpodServer> {
              * Verify the web socket handshake request.
              */
             const verifyClient: ws.VerifyClientCallbackAsync = async (info, callback) => {
-                if (!verifyCSRF(info.origin)) {
+                if (!verifyCSRF(info.origin, info.req.url === GitpodServerPath)) {
```

Comment:
`info.req.url === GitpodServerPath` is hard to understand in that context. why is it mapped to `strict`?

Reply:
```
export const GitpodServerPath = "/gitpod";
```

Reply:
why cannot it be always `true`?

Reply:
Because `/v1` should still work with bearer token.
But I don't think we still use/need this in workspace origin :thinking: cc @akosyakov

Reply:
check all clients, ie ides and local app if they work then we can generally make it more strict

Reply:
Bearer tokens need to be supported, it's how local app, or public-api connects with a Personal Access Token.

Reply:
While I think the changes make sense, I would like to push for being more strict, and making the changes more explicit in code.

I will push a suggestion on a different branch so we can compare.

Reply:
@geropl we are fine if you take it over 😊 just let us know that @mustard-mh or @jeanp413 don't look into it anymore.

Reply:
@akosyakov Ok, I will take over. Would appreciate help with testing it, though :upside_down_face: